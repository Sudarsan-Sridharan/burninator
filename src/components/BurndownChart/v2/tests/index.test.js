import React from 'react';
import { render } from 'testing-utils';

import V2Chart from '../';

import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { ThemeProvider } from 'styled-components/macro';

const customRender = (node, customState, ...options) => {
  const mockReducer = (state, action) => state;
  const initialState = {
    burndown: {
      currentBoard: {
        boardData: {
          processed: {
            enhancedIssueList: [
              // issue
              {
                isFinalInVersion: true,
                key: 'issue-1',
                pointsRemainingAfterMe: 1,
                version: 'v1',
              },
            ],
            v2ChartData: {
              outsideVersionRefLines: [],
              chartData: [],
            },
            velocityData: {
              overridden: false,
            },
            opts: {
              computedConfig: {
                scopeGrowth: {
                  isOverridden: false,
                },
              },
            },
          },
        },
      },
    },
  };
  const mockStore = createStore(mockReducer, { ...initialState, ...customState });
  const mockTheme = {};
  return render(
    <Provider store={mockStore}>
      <ThemeProvider theme={mockTheme}>{node}</ThemeProvider>
    </Provider>,
    ...options
  );
};

describe('<V2Chart />', () => {
  test('snapshot:issues', () => {
    const { container } = customRender(<V2Chart />, {});
    expect(container.firstChild).toMatchSnapshot();
  });
  test('snapshot:versions', () => {
    const { container } = customRender(<V2Chart display="versions" />, {});
    expect(container.firstChild).toMatchSnapshot();
  });
});
