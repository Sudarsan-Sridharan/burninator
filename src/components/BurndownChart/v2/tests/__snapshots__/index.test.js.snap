// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<V2Chart /> snapshot:issues 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 1.2;
  margin-bottom: 8px;
  text-transform: uppercase;
}

.c1 * {
  font-size: inherit;
}

.c2 * {
  font-size: inherit;
}

.c0 {
  border: 1px solid #ddd;
  background-color: #fff;
  padding: 8px 8px;
  margin-bottom: 8px;
  border-radius: 2px;
}

.c3 {
  margin: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

input.c4 {
  margin-top: 0;
  margin-bottom: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      undefined
    </div>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          class="c4"
          type="checkbox"
        />
        <span
          class="c5"
        >
          Tooltips
        </span>
      </label>
    </div>
  </div>
  <div
    class="recharts-responsive-container"
    style="width: 100%; height: 300px;"
  >
    <div
      style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
    />
  </div>
</div>
`;

exports[`<V2Chart /> snapshot:versions 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  line-height: 1.2;
  margin-bottom: 8px;
  text-transform: uppercase;
}

.c1 * {
  font-size: inherit;
}

.c2 * {
  font-size: inherit;
}

.c0 {
  border: 1px solid #ddd;
  background-color: #fff;
  padding: 8px 8px;
  margin-bottom: 8px;
  border-radius: 2px;
}

.c3 {
  margin: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

input.c4 {
  margin-top: 0;
  margin-bottom: 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div>
      undefined
    </div>
    <div
      class="c2"
    >
      <label
        class="c3"
      >
        <input
          class="c4"
          type="checkbox"
        />
        <span
          class="c5"
        >
          Tooltips
        </span>
      </label>
    </div>
  </div>
  <div
    class="recharts-responsive-container"
    style="width: 100%; height: 300px;"
  >
    <div
      style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
    />
  </div>
</div>
`;
